legit.py

```diff
@@ -1,54 +1,87 @@
 from git import Repo
 from git.exc import GitCommandError
 from typing import Iterable
+import logging
 
+# ==============================================================
+# create logger with 'legit'
+log = logging.getLogger("legit")
+log.setLevel(logging.DEBUG)
+# create file handler which logs even debug messages
+fh = logging.FileHandler("legit.log")
+fh.setLevel(logging.DEBUG)
+# create console handler which logs info messages
+ch = logging.StreamHandler()
+ch.setLevel(logging.ERROR)
+# create formatter and add it to the handlers
+formatter = logging.Formatter(
+    "%(asctime)s\t%(levelname)s -- %(processName)s %(filename)s:%(lineno)s -- %(message)s"
+)
+ch.setFormatter(formatter)
+fh.setFormatter(formatter)
+# add the handlers to the logger
+log.addHandler(ch)
+log.addHandler(fh)
+# ==============================================================
 
-class Repository:
+
+class Repository(object):
 
     def __init__(self, path: str) -> None:
         self.repo = Repo.init(path)
 
     @classmethod
     def clone(cls, url: str, path: str) -> str:
+        log.debug("Clone path", path)
         try:
             Repo.clone_from(url, path)
         except GitCommandError:
+            log.error("GitCommandError: no path")
```

Comment:
Maybe error message should be more informative. We make it for people.

Reply:
Maybe it's worth to write something like 
        except GitCommandError as git_error:
            log.error(f"GitCommandError: {git_error}")

Reply:
It depends of what this exception return as a message if it is not good enough let's use custom message